#! /bin/sh /usr/share/dpatch/dpatch-run
## 09_locale_parsing_fix-modules-gui-wxwidgets.dpatch by Daniel T Chen <crimsun@fungus.sh.nu>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Pull changeset 12732 from SVN trunk. Needed to fix locale parsing, 
## DP: which makes the wxWidgets playlist interface 'open' actually work

@DPATCH@

--- trunk/modules/gui/wxwidgets/interface.cpp	(revision 12731)
+++ trunk/modules/gui/wxwidgets/interface.cpp	(revision 12732)
@@ -1373,11 +1373,11 @@
 
     for( size_t i = 0; i < filenames.GetCount(); i++ )
     {
-        char *psz_utf8 = FromLocale( filenames[i].mb_str() );
+        char *psz_utf8 = FromUTF32( filenames[i].wc_str() );
         playlist_Add( p_playlist, psz_utf8, psz_utf8,
                       PLAYLIST_APPEND | ((i | b_enqueue) ? 0 : PLAYLIST_GO),
                       PLAYLIST_END );
-        LocaleFree( psz_utf8 );
+        free( psz_utf8 );
     }
 
     vlc_object_release( p_playlist );
--- trunk/modules/gui/wxwidgets/open.cpp	(revision 12731)
+++ trunk/modules/gui/wxwidgets/open.cpp	(revision 12732)
@@ -1147,17 +1147,17 @@
         playlist_item_t *p_item;
         char *psz_utf8;
 
-        psz_utf8 = FromLocale( mrl[i].mb_str() );
+        psz_utf8 = FromUTF32( mrl[i].wc_str() );
         p_item = playlist_ItemNew( p_intf, psz_utf8, psz_utf8 );
-        LocaleFree( psz_utf8 );
+        free( psz_utf8 );
 
         /* Insert options */
         while( i + 1 < (int)mrl.GetCount() &&
-               ((const char *)mrl[i + 1].mb_str())[0] == ':' )
+               ((const char *)mrl[i + 1].wc_str())[0] == ':' )
         {
-            psz_utf8 = FromLocale( mrl[i + 1].mb_str() );
+            psz_utf8 = FromUTF32( mrl[i + 1].wc_str() );
             playlist_ItemAddOption( p_item, psz_utf8 );
-            LocaleFree( psz_utf8 );
+            free( psz_utf8 );
             i++;
         }
 
@@ -1166,9 +1166,9 @@
         {
             for( int j = 0; j < (int)subsfile_mrl.GetCount(); j++ )
             {
-                psz_utf8 = FromLocale( subsfile_mrl[j].mb_str() );
+                psz_utf8 = FromUTF32( subsfile_mrl[j].wc_str() );
                 playlist_ItemAddOption( p_item, psz_utf8 );
-                LocaleFree( psz_utf8 );
+                free( psz_utf8 );
             }
         }
 
@@ -1177,9 +1177,9 @@
         {
             for( int j = 0; j < (int)sout_mrl.GetCount(); j++ )
             {
-                psz_utf8 = FromLocale( sout_mrl[j].mb_str() );
-                playlist_ItemAddOption( p_item, sout_mrl[j].mb_str() );
-                LocaleFree( psz_utf8 );
+                psz_utf8 = FromUTF32( sout_mrl[j].wc_str() );
+                playlist_ItemAddOption( p_item, psz_utf8 );
+                free( psz_utf8 );
             }
         }
 
--- trunk/modules/gui/wxwidgets/dialogs.cpp	(revision 12731)
+++ trunk/modules/gui/wxwidgets/dialogs.cpp	(revision 12732)
@@ -405,7 +405,7 @@
 
         for( size_t i = 0; i < paths.GetCount(); i++ )
         {
-            char *psz_utf8 = FromLocale( paths[i].mb_str() );
+            char *psz_utf8 = FromUTF32( paths[i].wc_str() );
             if( event.GetInt() )
                 playlist_Add( p_playlist, psz_utf8, psz_utf8,
                               PLAYLIST_APPEND | (i ? 0 : PLAYLIST_GO),
@@ -413,7 +413,7 @@
             else
                 playlist_Add( p_playlist, psz_utf8, psz_utf8,
                               PLAYLIST_APPEND, PLAYLIST_END );
-            LocaleFree( psz_utf8 );
+            free( psz_utf8 );
         }
     }
 
@@ -436,11 +436,11 @@
     if( p_dir_dialog && p_dir_dialog->ShowModal() == wxID_OK )
     {
         wxString path = p_dir_dialog->GetPath();
-        char *psz_utf8 = FromLocale( path.mb_str() );
+        char *psz_utf8 = FromUTF32( path.wc_str() );
         playlist_Add( p_playlist, psz_utf8, psz_utf8,
                       PLAYLIST_APPEND | (event.GetInt() ? PLAYLIST_GO : 0),
                       PLAYLIST_END );
-        LocaleFree( psz_utf8 );
+        free( psz_utf8 );
     }
 
     vlc_object_release( p_playlist );
